internal/cortex/querier/queryrange/query_range.go

```diff
@@ -184,6 +186,14 @@ func (resp *PrometheusInstantQueryResponse) GetStats() *PrometheusResponseStats
 	return resp.Data.Stats
 }
 
+func (resp *PrometheusResponse) AddHeader(key, value string) {
+	resp.Headers = append(resp.Headers, &PrometheusResponseHeader{Name: key, Values: []string{value}})
+}
+
+func (resp *PrometheusInstantQueryResponse) AddHeader(key, value string) {
+	resp.Headers = append(resp.Headers, &PrometheusResponseHeader{Name: key, Values: []string{value}})
+}
+
```

Comment:
I think we'll end up with duplicate headers if the same header key is added twice - should this check for the header already existing and either replace it or append to its values?